kiwi/xml_parse.py

```diff
@@ -5074,7 +5074,7 @@ def validate_portnum_type(self, value):
             if not self.gds_validate_simple_patterns(
                     self.validate_portnum_type_patterns_, value):
                 warnings_.warn('Value "%s" does not match xsd pattern restrictions: %s' % (value.encode('utf-8'), self.validate_portnum_type_patterns_, ))
-    validate_portnum_type_patterns_ = [['^(\\d+$|^\\d+/(udp$|^tcp))$']]
+    validate_portnum_type_patterns_ = [['^(\\d+$|\\d+/(udp|tcp))$']]
```

Comment:
is not this file autogenerated?

Reply:
yes the file is autogenerated. The regular expression is taken from https://github.com/OSInside/kiwi/blob/368f44262cb56c448b101a373003efca4a790a02/kiwi/schema/kiwi.rnc#L37

kiwi.rng is autogenerated too :)

Reply:
Yes the Relax schema is maintained in its condensed format (RNC) Thus schema changes must be performed on kiwi.rnc. Once a change there is made call:

```
make kiwi/schema/kiwi.rng
```

This builds the RNG schema as well as auto generates the data structures. The make call must be performed from within a tox development environment. For details read:

* http://osinside.github.io/kiwi/development.html#create-a-python-virtual-development-environment

Thanks

Reply:
> is not this file autogenerated?

Ah, ok. Didn't know that.